clang/lib/Parse/ParseDecl.cpp

```diff
@@ -764,6 +764,73 @@ void Parser::ParseGNUAttributeArgs(
                            ScopeLoc, Form);
 }
 
+void Parser::ParseAtomicAttribute(
```

Comment:
I'm very much beginning to dislike the syntax for this, if it needs this much special handling.  We shoudl find a way to spell this attribute in a way that WORKS with our existing attribute formats.  Else this becomes REALLY difficult for other compilers to ignore.